transformations/suspecting_paraphraser/transformation.py

```diff
@@ -41,72 +42,78 @@ def __init__(self, seed=0, max_output=1, pronoun_mod=0.9):
+        verb_position = [i for i in range(len(doc)) if str(doc[i]) == token.head.text][
+            0
+        ]
 
         rest_of_sentence = [i for i in text[verb_position:]]
+
+        if text[1].lower() == "n't":
+            text = text[1:]
+            doc = doc[1:]
+            verb_position = verb_position - 1
+
         beginning = [text[1].capitalize()]
         beginning.extend(text[2:verb_position])
-        sentence = nltk.tokenize.treebank.TreebankWordDetokenizer().detokenize(beginning + rest_of_sentence)
+        sentence = nltk.tokenize.treebank.TreebankWordDetokenizer().detokenize(
+            beginning + rest_of_sentence
+        )
 
         first_verb = doc[verb_position]
-        if modal=='did':
-            demodded = first_verb._.inflect('VBD')
+        if modal == "did":
+            demodded = first_verb._.inflect("VBD")
         else:
             demodded = modal + " " + str(first_verb)
         sentence = sentence.replace(str(first_verb), str(demodded)).replace("?", "")
 
         ending = self._resolve_ending(doc, modal)
         result = sentence + ending
-        print(result)
         return result
 
     def _resolve_ending(self, doc, modal):
         try:
-            subject = str([tok for tok in doc if (tok.dep_ == "nsubj") ][0])
+            subject = str([tok for tok in doc if (tok.dep_ == "nsubj")][0])
         except IndexError:
             print(doc)
             return ", right?"
 
-        prob = np.ones(len(self.pronouns))/len(self.pronouns)
+        prob = np.ones(len(self.pronouns)) / len(self.pronouns)
 
         tagged = [(X.text, X.label_) for X in doc.ents]
         if subject.lower() in self.static_pronouns:
             pronoun = subject.lower()
             if pronoun == "i":
                 pronoun = "I"
         else:
-            if len(tagged) > 0 and tagged[0][1]!='PERSON':
-                prob = [0]*len(self.pronouns)
+            if len(tagged) > 0 and tagged[0][1] != "PERSON":
+                prob = [0] * len(self.pronouns)
                 prob[Pronouns.IT] = 1
             else:
                 noun_gender = self.gender_detector.get_gender(subject)
 
-                if noun_gender in ['male', 'mostly_male']:
+                if noun_gender in ["male", "mostly_male"]:
                     prob = [self._pronoun_alt] * len(self.pronouns)
                     prob[Pronouns.HE] = self.pronoun_mod
 
-                elif noun_gender in ['female', 'mostly_female']:
+                elif noun_gender in ["female", "mostly_female"]:
                     prob = [self._pronoun_alt] * len(self.pronouns)
                     prob[Pronouns.SHE] = self.pronoun_mod
 
             pronoun = np.random.choice(self.pronouns, p=prob)
-        if modal=='have' and (pronoun in ['he', 'she', 'it']):
-            modal = 'has'
+        if modal == "have" and (pronoun in ["he", "she", "it"]):
+            modal = "has"
```

Comment:
Can would, should and could also be potential modals to handle here?

Reply:
I've deleted this section altogether as it was redundant